native/general_double.go

```diff
@@ -10,6 +10,21 @@ import (
 	"math"
 )
 
+func checkMatrix(m, n int, a []float64, lda int) {
```

Comment:
Rename this checkMatrix64, then you can generate the float32 version and change all the calls to be appropriate.